core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -583,7 +586,15 @@ angular.module('oppia').directive('questionPlayer', [
               ctrl.testIsPassed = hasUserPassedTest();
             }
           });
-        }
-      ]
+          ctrl.$onInit = function() {
+            ctrl.userIsLoggedIn = null;
+            UserService.getUserInfoAsync().then(function(userInfo) {
+              ctrl.canCreateCollections = userInfo.canCreateCollections();
+              ctrl.userIsLoggedIn = userInfo.isLoggedIn();
+            });
+            initResults();
```

Comment:
If initResults() is not being called from elsewhere, maybe inline its contents here.

Reply:
Added comment.